src/DominantJuice.sol

```diff
@@ -0,0 +1,399 @@
+     * call chain described in `redeemParams()`.
+     * @param _data Standard Juicebox project redemption data.
+     * See https://docs.juicebox.money/dev/api/data-structures/jbdidredeemdata3_1_1/.
+     */
+    function didRedeem(JBDidRedeemData3_1_1 calldata _data) external payable terminalCheck redeemCycleCheck {
+        if (_data.projectId != campaign.projectId) revert IncorrectProjectID();
+
+        address payable redeemer = payable(_data.holder);
+        if (unwrap(pledgers.pledgerWeight[redeemer]) == 0) revert MustBePledger();
+        if (pledgers.hasBeenRefunded[redeemer]) revert AlreadyWithdrawnRefund();
+
+        // Calculates the refund bonus, using PRBMath, based on the time from cycle start and the size of the pledge
+        uint256 pledgerRefundBonus = unwrap(
+            mul(div(pledgers.pledgerWeight[redeemer], pledgers.totalPledgeWeight), wrap(campaign.totalRefundBonus))
+        );
+
+        // Sanity Check
+        if (address(this).balance < pledgerRefundBonus) revert InsufficientFunds();
+
+        // This redeeem function is now locked for the current redeeming address.
+        pledgers.hasBeenRefunded[redeemer] = true;
+
+        // Sending early pledger refund bonus.
+        (bool sendSuccess,) = redeemer.call{value: pledgerRefundBonus}("");
+        require(sendSuccess, "Failed to send refund bonus.");
+
+        emit CycleRefundBonusWithdrawal(redeemer, pledgerRefundBonus);
+    }
+
+    /**
+     * @notice This function is callable after cycleExpiryDate if goal is met. However, for
+     * contingency, where a redeemer is unable to retrieve funds, after a locked period, the
+     * Campaign Manager can withdraw funds for disbursement to remaining redeemers . The unlock
+     * time is hardcoded for two weeks after the cycleExpiryDate to give pledgers confidence
+     * that funds are programmatically guaranteed to remain in the contract for that time.
+     * @param receivingAddress Campaign Manager must call this function but can withdraw funds to
+     * another address if desired.
+     */
+    function creatorWithdraw(address payable receivingAddress, uint256 amount)
+        external
+        payable
+        onlyRole(CAMPAIGN_MANAGER_ROLE)
+    {
+        uint256 unlockTime = campaign.cycleExpiryDate + 1209600; // Two weeks after expiry date
+        require(
+            (hasCycleExpired() && isTargetMet()) || ((block.timestamp > unlockTime) && !isTargetMet()),
+            "Cycle must be expired and successful, or it must be passed the lock period."
+        );
+        if (amount > address(this).balance) revert InsufficientFunds();
+
+        (bool success,) = receivingAddress.call{value: amount}("");
+        require(success, "Failed to withdraw cycle funds.");
+
+        emit CreatorWithdrawal(receivingAddress, amount);
+    }
+
+    /// Getters and Helpers
+
+    function getCampaignInfo() public view returns (Campaign memory) {
```

Comment:
do you need this? Now that you have a Campaign struct, you could just make that variable public if you are going to create the getter anyway

Reply:
I thought you wanted all variables private? I’m getting mixed signals here, Michael! Haha.

Reply:
sorry, I was confusing. If you want the value to be part of the API, then you make it public. I was trying to say that you should be judicious in your use of public. If you are just going to create a getter anyway, then that's what public does for you. So...

If you are going to create a getter for a field (ie you want it to be part of the public API), then make the field public.

If you would not make a getter for the field (ie you do not want it to be part of the public API), then make the field private or internal.

Reply:
Got it, thanks for clarification.